src/apps/chifra/pkg/types/types_trace.go

```diff
@@ -204,7 +204,10 @@ func mustParseUint(input any) (result uint64) {
 }
 
 func (s *SimpleTrace) MarshalCache(writer io.Writer) (err error) {
-	if err = cacheNew.WriteValue(writer, s.Action); err != nil {
+	optAction := &cacheNew.Optional[SimpleTraceAction]{
```

Comment:
I think I understand what this does, but can you provide a short explanation? Trace is one of the few things (along with Transaction which has Receipt) that has fields that are objects in their own right (Action and Result). This acts like a wrapper so the Read and Write code is simpler?

Reply:
It's because Action and Result are pointers, so they can be nil. In such a case, we want to store information, that there's no value to read. I'll add a comment 👍

Reply:
Let's discuss.